Sources/Masse/PodcastEntry.swift

```diff
@@ -16,12 +16,30 @@ struct PodcastEntry {
     let date: Date
     let filename: String
     
-    init(meta: [String: String], filename: String) {
+    init(meta: [String: String], filename: String, folder: String) {
         self.meta = meta
         self.filename = filename
         let dateString = meta[Keys.PodcastEntry.date.rawValue].expect("Need \(Keys.PodcastEntry.date.rawValue) entry in podcast entry \(filename)")
         date = PodcastEntry.dateFormatter.date(from: dateString).expect("Invalid formatted date \(dateString) in post \(filename)")
+        
         // insert the podcast-format date into the meta
         self.meta["podcastDate"] = PodcastEntry.podcastDateFormatter.string(from: date)
+        
+        // calculate the duration
+        if let mp3Filename = meta[Keys.PodcastEntry.file.rawValue] {
+            let url = URL(fileURLWithPath: "\(folder)/\(mp3Filename)")
+            if let data = try? Data(contentsOf: url) {
+                self.meta[Keys.PodcastEntry.length.rawValue] = "\(data.count)"
+                let stream = InputStream(data: data)
+                do {
+                    stream.open()
+                    let calculator = MP3DurationCalculator(inputStream: stream)
+                    let duration = try calculator.calculateDuration()
+                    self.meta[Keys.PodcastEntry.duration.rawValue] = "\(duration)"
```

Comment:
`duration.description` would be more stylistic (and potentially more performant), I suppose?